### ISSUE TITLE
keras early callback stopping stage question

### ISSUE THREAD

**Issue by raun1** [fchollet/keras#6596]
Hello,

I just want to verify if the keras early call back retains the weight at the best network performance or does it retain the weight after the end of the patience?
i.e. if my network at x epoch had 80% validation accuracy and at x+3 epoch it stops training but validation accuracy is now 70%. 

Does keras retain the weights at the epoch x or at epoch x+3?

- [- ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ -] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
